src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -148,6 +149,9 @@ async Task EnsureLong()
                 }
                 catch (Exception e)
                 {
+                    if (e is OperationCanceledException)
```

Comment:
We can use an exception filter (catch ... when...) rather than catching exception and checking inside for OperationCanceledException.